Feature/ab#37 legal guardian controller

vyruz1986 opened the pull request.

vyruz1986 committed:
feat(LegalGuardianUserController): Implemented controller with mock data

This is using mockdata, models & service still to be implemented

vyruz1986 committed:
Merge branch 'develop' into feature/AB#37-legal-guardian-controller

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

vyruz1986 committed:
Merge branch 'develop' into feature/AB#37-legal-guardian-controller

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

vyruz1986 committed:
Merge remote-tracking branch 'origin/develop' into feature/AB#37-legal-guardian-controller

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

vyruz1986 committed:
Merge remote-tracking branch 'origin/develop' into feature/AB#37-legal-guardian-controller

vyruz1986 committed:
fix: Fixeth the build

Joren-Thijs committed:
feat(legalguardians) Made API service and models
with dummy data. Also moved pagination model to own file.

Joren-Thijs committed:
feat(legalguardians) generated mat table
and added imports to material.module.
Moved mat-date-adapter into material.module as well

AB#37

Joren-Thijs committed:
feat(legalguardians) reformatted table to specs
Added required fields
Added row selectiona
added add button
Moved pagination into  table
Added placeholder methods for selection and adding
Added data from API
Implemented back-end type data as far as possible
Implemented observables
Added filter and sorting events
Added table styling
Added routing
Added nav-menu button

AB#37

Joren-Thijs committed:
feat(legalguardian) generated detail component

AB#37

Joren-Thijs committed:
feat(legalguardian) Created legal guardian form

AB#37

Joren-Thijs committed:
feat(legalguardian) added form opening to overview
+ bug fixes in legal-guardian-details-component

AB#37

Joren-Thijs committed:
feat(legalguardian) fixed overview table styling

disabled white-space wrapping

AB#37

vyruz1986 committed:
feat(LegalGuardianUsers): Added updated user model from dev

vyruz1986 committed:
Merge branch 'feature/AB#37-legal-guardian-controller' of github.com:Brixel/Singer into feature/AB#37-legal-guardian-controller

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

vyruz1986 committed:
feat(LegalGuardianUser): Added DBContext for LG Users

Joren-Thijs committed:
feat(legalguardian) fixed table width in CSS

AB#37

Joren-Thijs committed:
feat(legalguardian) merge

AB#37

vyruz1986 committed:
fix(LegalGuardianUsers): Reverted faulty migration

vyruz1986 committed:
feat(LegalGuardianUsers): Fixed migrations, implemented service & controller

This commit corrects the migrations for the LegalGuardianUsers,
adds the LegalGuardianUserService
and the LegalGuardianUserController

vyruz1986 committed:
feat: Seed LegalGuardianUsers

vyruz1986 committed:
fix(DataControllerBase): Fixed pagination in base controller

The page numbers were calculated based on row number, these are now
using page numbers (as intended)

vyruz1986 committed:
fix(LegalGuardianUSers): Added missing fields to DTO

vyruz1986 committed:
feat(LegalGuardianUsers): Added  map from LegalGuardianDTO to LegalGuardian

wim07101993 committed:
refactor(service-abstraction): changed the parameter type of the create methods to TCreateDTO

vyruz1986 committed:
Merge remote-tracking branch 'origin/feature/service-abstraction' into feature/AB#37-legal-guardian-controller

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

vyruz1986 committed:
fix: Fixed TCreateDTO references

vyruz1986 committed:
feat(UserService): Implemented abstract UserService

This replaces CareUSerService, which has been renamed as such

vyruz1986 committed:
refactor(Evereything): Models & DTOs refactored to make more sense

Godspeed to all who need to resolve conflicts

vyruz1986 committed:
refactor(UserService): Added automapper conf, improved createAsync

vyruz1986 committed:
fix(LGUserService): Implemented custom mapper to fill in user fields

This should be handled by the Automapper, but couldn't get that to work,
so we'll use this custom mapper for now

vyruz1986 committed:
fix(AutomapperProfiles): Attempt at fixing automapper profiles

This is not yet working, but at least it's better than it was

vyruz1986 committed:
fix(CareUserController): Added default values for GetUsers

Also fixed PageIndex numbering

vyruz1986 committed:
fix(UserService): Fixed GetOneAsync

vyruz1986 committed:
fix(DabaseService): Fixed UpdateAsync

Detach the local item before trying to update it.

BerendWouters committed:
feat(user) extends IUser with User property

AB#37

BerendWouters committed:
chore cleanup some of the userservice

AB#37

vyruz1986 committed:
fix(UserService): Fix deleting of User

Also added protection against invalid id in GetOneAsync

vyruz1986 committed:
Merge remote-tracking branch 'origin/feature/AB#37-legal-guardian-controller' into feature/AB#37-legal-guardian-controller

BerendWouters committed:
chore introduce Queryable property on the DatabaseService

AB#37

BerendWouters committed:
Merge branch 'feature/AB#37-legal-guardian-controller' of https://github.com/Brixel/Singer into feature/AB#37-legal-guardian-controller

vyruz1986 committed:
fix(UserService): Use UserId instead of Id to delete user

vyruz1986 committed:
Merge remote-tracking branch 'origin/develop' into feature/AB#37-legal-guardian-controller

Joren-Thijs committed:
feat(legalguardian)  implement legalguardian proxy

AB#37

vyruz1986 committed:
fix(DataControllerBase): Added FromBody to Create and Update function

This ensures that the body will be mapped to the DTO

BerendWouters commented on Singer.API/ClientApp/src/app/modules/admin/admin.module.ts:
```diff
@@ -10,29 +10,33 @@ import { AgegroupPipe } from '../core/services/agegroup.pipe';
       ApiService,
       AgegroupPipe,
       KeysPipe,
-      { provide: MAT_MOMENT_DATE_ADAPTER_OPTIONS, useValue: { useUtc: true } }]
```
Where's this provide gone?

vyruz1986 committed:
fix(Angular): Fixed field names of LegalGuardians

Removed username & birthdate
Added postal code, city and country

vyruz1986 committed:
fix(LegalGuardians): Fixed filtering/sorting (somewhat)

Re-added searchParams to HTTP get call
Added translation for 'asc' and 'desc' sort directions in
DataControllerBase
Sorting on LegalGuardians firstname/lastname/email is not yet working

vyruz1986 committed:
fix(AdminModule): Fixed merge issue which nerfed MatMomentDateModule
